### PR TITLE
[VEGA-3237] ci: add common ci build

### PR DESCRIPTION
**Pull request by ababakov**
### Description
Добавлены sh скрипты общей сборки и образ сборщика 

### Checks
- [x] Jira link attached: [VEGA-3237](https://jira.vegaspace.tk/browse/VEGA-3237)
- [ ] Possible Associated Jira links:
- [ ] Personal Vega Instance link attached: http://xx-yy-vega-builder.code013.org
- [ ] Jira tasks for technical debt were created (if needed)

### Type
- [ ] Feature
- [ ] Alpha Feature ([Alpha] MR/PR Title)
- [ ] Bug Fix
- [ ] Test
- [ ] Refactoring

### PR/MR Requirements
- [x] PR/MR title meet requirements/convention
- [ ] Target branch is correct
- [ ] PR/MR branch was rebased at correct branch
- [ ] Diff was checked
- [ ] Configuration files were checked at test repositories
- [x] [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)

### General
- [ ] Exploratory Tested at Personal Vega Instance
- [ ] Existing E2E Tests done, Screenshot attached
- [ ] Existing Integration Tests done, Screenshot attached, Jenkins link:
- [ ] Existing Unit Tests done, Screenshot attached
- [ ] New E2E Tests developed
- [ ] New Integration Tests developed
- [ ] New Unit Tests developed
- [ ] API documentation was fully and correctly updated

### E2E Suite Jenkins link
_{link}_

### E2E Suite Screenshot
_{Screenshot}_

### Integration Tests Screenshot
_{Screenshot}_

### Unit Tests Screenshot
_{Screenshot}_

### CODE REVIEW

**Review by d4rkm3z — APPROVED** [review]

File: ci/build-entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
```
Comment (alexeyandriyash) [inline ci/build-entrypoint.sh:2]: set -e

File: ci/build.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
```
Comment (alexeyandriyash) [inline ci/build.sh:2]: set -e

File: ci/build-entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+NPMRC_TEMP=$(cat .npmrc)
+
+sed -e "s/\$NPM_URI/$NPM_URI/" \
+    -e "s/\$NPM_AUTH_TOKEN/$NPM_AUTH_TOKEN/" ./ci/npmrc-template > .npmrc
+
+yarn install --frozen-lockfile
+yarn build:shell
+
+echo -e "$NPMRC_TEMP" > .npmrc
```
Comment (alexeyandriyash) [inline ci/build-entrypoint.sh:22]: left the comment "rollback .npmrc"

Comment (alexeyandriyash) [reply ci/build-entrypoint.sh:22]: better to use bash traps

File: ci/build-entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  echo "NPM_AUTH_TOKEN is required to continue. Abort."
+  exit 1;
+fi
+
+NPMRC_TEMP=$(cat .npmrc)
+
+sed -e "s/\$NPM_URI/$NPM_URI/" \
+    -e "s/\$NPM_AUTH_TOKEN/$NPM_AUTH_TOKEN/" ./ci/npmrc-template > .npmrc
+
+yarn install --frozen-lockfile
```
Comment (alexeyandriyash) [inline ci/build-entrypoint.sh:19]: this could be a separated script, other parts are all-frontend common

File: ci/build-entrypoint.sh [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+yarn build:shell
```
Comment (alexeyandriyash) [inline ci/build-entrypoint.sh:20]: this could be a separated script, other parts are all-frontend common

**Review by alexeyandriyash — APPROVED** [review]

**Pull request merged**